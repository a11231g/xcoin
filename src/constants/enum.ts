export enum CURRENCY {
  USD = 'USD',
  GBP = 'GBP',
  EUR = 'EUR'
}

export enum SYMBOL {
  USD = '$',
  GBP = '€',
  EUR = '£'
}

export enum RATE_OPERATOR {
  minus = 'minus',
  plus = 'plus'
}
