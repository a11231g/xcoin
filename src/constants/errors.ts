export const ERROR = {
  noEnoughMoney: 'no enough balance'
}
