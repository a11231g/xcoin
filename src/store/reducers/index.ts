import { combineReducers } from '@reduxjs/toolkit'

import rehydrate from './rehydrate.reducer'
import inputs from './inputs.reducer'
import balance from './balance.reducer'
import rate from './rate.reducer'

const rootReducer = combineReducers({
  rehydrate,
  inputs,
  balance,
  rate
})

export default rootReducer
