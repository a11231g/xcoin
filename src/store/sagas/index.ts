import rehydrate from './rehydrate.saga'
import rate from './rate.saga'
import input from './inputs.saga'
import change from './balance.saga'

const sagas = [...rehydrate, ...rate, ...input, ...change]

export default sagas
