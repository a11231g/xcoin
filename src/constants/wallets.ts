import { CURRENCY, SYMBOL } from './enum'

export const WALLETS = [
  { label: CURRENCY.USD, value: CURRENCY.USD, symbolTxt: SYMBOL.USD },
  { label: CURRENCY.EUR, value: CURRENCY.EUR, symbolTxt: SYMBOL.EUR },
  { label: CURRENCY.GBP, value: CURRENCY.GBP, symbolTxt: SYMBOL.GBP }
]
