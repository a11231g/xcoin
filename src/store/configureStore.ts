import { createStore, applyMiddleware, compose } from 'redux'
import createSagaMiddleware from 'redux-saga'
import { offline } from '@redux-offline/redux-offline'
import offlineConfig from '@redux-offline/redux-offline/lib/defaults'
import { createBlacklistFilter } from 'redux-persist-transform-filter'
import { startRehydrate } from './reducers/rehydrate.reducer'
import rootReducer from './reducers'
import sagas from './sagas'
import { Config as ReduxOfflineConfig } from '@redux-offline/redux-offline/lib/types'

function configureStore(initialState = {}, persistOffline = true) {
  const sagaMiddleware = createSagaMiddleware()

  let middlewares = applyMiddleware(sagaMiddleware)
  if (persistOffline) {
    const amendedConfig: Partial<ReduxOfflineConfig> = {
      ...offlineConfig,

      persistOptions: {
        whitelist: ['inputs', 'balance', 'rate'],
        transforms: [createBlacklistFilter('inputs', ['firstInputValue', 'secondInputValue'])]
      },
      persistCallback: () => store.dispatch(startRehydrate())
    }
    const composeEnhancers =
      (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose
    middlewares = composeEnhancers(middlewares, offline(amendedConfig))
  }
  const store = createStore(rootReducer, initialState, middlewares)

  sagas.map(saga => sagaMiddleware.run(saga, store.dispatch))
  return store
}

const store = configureStore({}, true)

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export default store
