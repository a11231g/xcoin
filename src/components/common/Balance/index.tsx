import React from 'react'
import { StyleSheet, Text } from 'react-native'
import { useAppSelector } from 'store'
import colors from 'style/colors'
import { CurrenyItem } from 'types/types'

type Props = {
  selected: CurrenyItem
}
function Balance(props: Props) {
  const { selected } = props
  const balance = useAppSelector(state => state.balance)

  return (
    <Text testID={'balance-text'} style={styles.text}>
      Balance: {`${balance[selected.value]} ${selected?.symbolTxt}`}
    </Text>
  )
}

const styles = StyleSheet.create({
  text: {
    color: colors.gunMetal,
    fontSize: 14,
    fontWeight: '100',
    maxWidth: 160
  }
})

export default Balance
