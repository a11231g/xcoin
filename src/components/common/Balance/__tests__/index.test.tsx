import React from 'react'
import renderer from 'react-test-renderer'

import Balance from '../index'
import { CURRENCY, SYMBOL } from 'constants/enum'
import { Provider } from 'react-redux'
import store from 'store/configureStore'

function getValueByTestID(testRenderer: any, testID: string) {
  return testRenderer.root.findByProps({ testID }).props.children
}

describe('<Balance />', () => {
  it('renders correctly', () => {
    const selected = {
      value: CURRENCY.USD,
      label: CURRENCY.USD,
      symbolTxt: SYMBOL.USD
    }
    const testRenderer = renderer.create(
      <Provider store={store}>
        <Balance selected={selected} />
      </Provider>
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()

    expect(getValueByTestID(testRenderer, 'balance-text')).toEqual(['Balance: ', '200 $'])
  })
})
