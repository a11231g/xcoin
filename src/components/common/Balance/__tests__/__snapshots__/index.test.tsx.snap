// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Balance /> renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#4e5154",
      "fontSize": 14,
      "fontWeight": "100",
      "maxWidth": 160,
    }
  }
  testID="balance-text"
>
  Balance: 
  200 $
</Text>
`;
